test/testdata/parser/error_recovery/unop_binop.rb

```diff
@@ -0,0 +1,68 @@
+    f = T.let(false, T::Boolean)
+    puts (0 .. def) # error: unexpected token ")"
+    puts (1 ... def) # error: unexpected token ")"
+    puts (2 + def) # error: unexpected token ")"
+    puts (2 +) # error: unexpected token ")"
+    puts (3 - def) # error: unexpected token ")"
+    puts (3 -) # error: unexpected token ")"
+    puts (4 * def) # error: unexpected token ")"
+    puts (4 *) # error: unexpected token ")"
+    puts (5 / def) # error: unexpected token ")"
+    puts (5 /) # error: unexpected token ")"
+    puts (6 % def) # error: unexpected token ")"
+    puts (6 % ) # error: unexpected token ")"
+    puts (7 ** def) # error: unexpected token ")"
+    puts (7 **) # error: unexpected token ")"
+    puts (-8 ** def) # error: unexpected token ")"
+    puts (-8 **) # error: unexpected token ")"
+    puts (+9 ** def) # error: unexpected token ")"
+    puts (+9 **) # error: unexpected token ")"
+    puts (-def) # error: unexpected token ")"
+    puts (-) # error: unexpected token ")"
+    puts (+def) # error: unexpected token ")"
+    puts (+) # error: unexpected token ")"
+    puts (10 | def) # error: unexpected token ")"
+    puts (10 |) # error: unexpected token ")"
+    puts (11 ^ def) # error: unexpected token ")"
+    puts (11 ^) # error: unexpected token ")"
+    puts (12 & def) # error: unexpected token ")"
+    puts (12 &) # error: unexpected token ")"
+    puts (13 <=> def) # error: unexpected token ")"
+    puts (13 <=>) # error: unexpected token ")"
+    puts (14 == def) # error: unexpected token ")"
+    puts (14 ==) # error: unexpected token ")"
+    puts (15 === def) # error: unexpected token ")"
+    puts (15 ===) # error: unexpected token ")"
+    puts (16 != def) # error: unexpected token ")"
+    puts (16 !=) # error: unexpected token ")"
+    puts (/17/ =~ def) # error: unexpected token ")"
+    puts (/17/ =~) # error: unexpected token ")"
+    puts (/18/ !~ def) # error: unexpected token ")"
+    puts (/18/ !~) # error: unexpected token ")"
+    puts (!def) # error: unexpected token ")"
+    puts (!) # error: unexpected token ")"
+    puts (~def) # error: unexpected token ")"
+    puts (~) # error: unexpected token ")"
+    puts (20 << def) # error: unexpected token ")"
+    puts (20 <<) # error: unexpected token ")"
+    puts (21 >> def) # error: unexpected token ")"
+    puts (21 >>) # error: unexpected token ")"
+    puts (t && def) # error: unexpected token ")"
+    puts (t &&) # error: unexpected token ")"
+    puts (f || def) # error: unexpected token ")"
+    puts (f ||) # error: unexpected token ")"
+    puts (24 > def) # error: unexpected token ")"
+    puts (24 >) # error: unexpected token ")"
+    puts (25 < def) # error: unexpected token ")"
+    puts (25 <) # error: unexpected token ")"
+    puts (26 >= def) # error: unexpected token ")"
+    puts (26 >=) # error: unexpected token ")"
+    puts (27 <= def) # error: unexpected token ")"
+    puts (27 <=) # error: unexpected token ")"
+  end
+end
```

Comment:
Can you add a test in `test/testdata/lsp/completion/` too? It doesn't have to cover all these cases, by any means, but it'd be nice to see a test for this:

```ruby
class A
  def foo(a, b, x, y)
    if x && 
      #     ^ completion: a, b, x, y, ...
    end
end
```

Reply:
i was thinking it would work out of the box. if you want to look into why, feel free, otherwise commit the test and make a ticket?

Reply:
Added a test that expects the current behavior, with a note about what we really want.